Sources/NetworkService/InternalError/NetworkServiceInternalError.swift

```diff
@@ -1,6 +1,5 @@
-import Foundation
 /// Defines a set of internal errors that can occur during using the `NetworkService`.
-public enum InternalError: Error {
+public enum NetworkServiceInternalError: Error, CaseIterable {
     /// Represents a failure during serialization or deserialization operations.
     case serializationFailure
 }
```

Comment:
```suggestion

    var localizedDescription: String {
        switch self {
        case .serializationFailure:
            return "The provided data can not be serialized."
        }
    }
}
```
We can only add `Error` protocol and provide `localizedDescription` property?
WDYT?

Reply:
I agree, that it can be easy to read and understand but I used the same style for describing the error as we used for the NetworkError to follow one style.

Reply:
Fixed with https://github.com/ios-course/simple-network-service/pull/13/commits/d1dba3ccd838cb7ed5f29b3aafb42d33df4c0d40